src/analysis/rust_type.rs

```diff
@@ -19,67 +19,45 @@ pub trait ToRustType {
     fn to_rust_type(&self) -> Result;
 }
 
-impl ToRustType for library::Fundamental {
```

Comment:
This impl seems good as is, it doesn't depend on the library and namespace and you're right to be concerned about the method below growing too big.